### PR TITLE
Install debian and python dependencies for LMS/CMS

### PR DESCRIPTION
**Pull request by jrbl**
- Create a global variable lms_source_repo to make it easy to update
  where to pull mitx from when mitx moves
- Create global variable venv_base to store location of edx virtual
  environment, defaults to /opt/edx
- Create edx virtual environment to store python modules
- Add tags to LMS and CMS tasks to make selective running and testing
  easier
- Install a long, static list of debian packages, which has been
  manually curated from other edx configurations; this probably needs
  more attention and in the future should probably be divided better
  among various roles.
- Install the list of python packages distributed with the mitx
  repository
- Install some really useful software, namely mosh and tmux

### CODE REVIEW

**Comment by jrbl** [comment]
oops, merge conflict. I'll resolve that now.

File: playbooks/group_vars/all [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ---
 app_base_dir: /opt/wwc
+venv_dir: /opt/edx
+lms_source_repo: git@github.com:MITx/mitx.git
```
Comment (jarv) [inline playbooks/group_vars/all]: I think we should put `lms_source_repo` in the lms role vars/main.yml

File: playbooks/roles/lms/tasks/main.yml [hunk 2]
```diff
@@ -17,18 +22,165 @@
...
+  - puppetmaster
+  - puppetmaster-common
+  - puppet-lint
+  - vim-puppet
+  - python-nosexcover
+  - libwww-perl
+  - libswitch-perl
+  - libcrypt-ssleay-perl
+  - libnet-amazon-ec2-perl
+  - github-cli
```
Comment (jarv) [inline playbooks/roles/lms/tasks/main.yml]: Can we move the list to roles/lms/vars/main.yml?

File: playbooks/group_vars/all (hunk 1 above)
Comment (jrbl) [inline playbooks/group_vars/all]: Ok, I'll move that now.

**Comment by jrbl** [comment]
I resolved the merge conflict and rebased on current master. Should be a bit nicer with the apt list alphabetized. I'll make the suggested change now and re-push in a few minutes.

File: playbooks/group_vars/all [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 app_base_dir: /opt/wwc
+venv_dir: /opt/edx
```
Comment (jarv) [inline playbooks/group_vars/all:3]: I'm thinking `app_base_dir` and `venv_dir` should move to the common role but we can do that later once we figure out the secure_dir stuff.

Comment (jrbl) [inline playbooks/group_vars/all:3]: Yeah, I think you should make a trello so we remember to do this, and then let's put it off.  I think some of the other things belong in common ultimately too.

**Pull request merged**